<launch>
  <arg name="robot_name"/>
  <arg name="init_pose"/>

  <group ns="$(arg robot_name)">
    <arg name="model" default="$(find ic120_description)/urdf/ic120_unity.xacro"/>
    <param name="robot_description" command="$(find xacro)/xacro $(arg model)"/>
    <param name="tf_prefix" value="$(arg robot_name)_tf" />

    <node name="odom_broadcaster" pkg="ic120_unity" type="odom_broadcaster.py" output="screen">
      <param name="odom_frame" value="$(arg robot_name)_tf/odom"/>
      <param name="base_link_frame" value="$(arg robot_name)_tf/base_link" />
    </node>

    <node name="poseStamped2ground_truth_odom" pkg="ic120_unity" type="poseStamped2Odometry.py" output="screen">
      <!-- <param name="odom_frame" value="$(arg robot_name)_tf/odom"/>
      <param name="base_link_frame" value="$(arg robot_name)_tf/base_link" /> -->
    </node>

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

    <node pkg="robot_localization" type="ekf_localization_node" name="ekf_global" clear_params="true" output="screen">
      <rosparam command="load" file="$(find ic120_unity)/config/ic120_ekf.yaml"/>
      <param name="tf_prefix" value=""/>
      <param name="map_frame" value="map"/>
      <param name="world_frame" value="map"/>
      <param name="odom_frame" value="$(arg robot_name)_tf/odom"/>
      <param name="base_link_frame" value="$(arg robot_name)_tf/base_link"/>
      <remap from="odometry/filtered" to="/$(arg robot_name)/odometry/global" />
      <remap from="odom0" to="/$(arg robot_name)/odom" />
      <remap from="odom1" to="/$(arg robot_name)/tracking/ground_truth" />
    </node>

    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
      <rosparam file="$(find ic120_nav)/params/costmap_common_params.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find ic120_nav)/params/costmap_common_params.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find ic120_nav)/params/odom_nav_params/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find ic120_nav)/params/odom_nav_params/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find ic120_nav)/params/base_local_planner_params.yaml" command="load" />
      <rosparam file="$(find ic120_nav)/params/move_base_params.yaml" command="load" />
      <param name="base_global_planner" type="string" value="global_planner/GlobalPlanner" />
      <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS"/>
      <param name="global_costmap/global_frame" value="map"/>
      <param name="global_costmap/robot_base_frame" value="$(arg robot_name)_tf/base_link"/>
      <param name="local_costmap/global_frame" value="map"/>
      <param name="local_costmap/robot_base_frame" value="$(arg robot_name)_tf/base_link"/>

      <remap from="cmd_vel" to="tracks/cmd_vel" />
      <remap from="odom" to="odom" /> 
    </node>
  </group>
</launch>
